File: .copilot/skills/workflow/examples/loop-until-dry.cwf.py
question = args or "Find likely issues in this project."
seen = set()
dry_rounds = 0


def find_more(round_index):
    global dry_rounds
    result = wf.agent(
        "Find new issues not already seen.\n\n"
        f"Question: {question}\n\nAlready seen:\n{sorted(seen)}",
        model="claude-haiku-4.5",
        phase="discover",
        label=f"round-{round_index}",
    )
    candidates = [line.strip() for line in result.content.splitlines() if line.strip()]
    new = [line for line in candidates if line not in seen]
    seen.update(new)
    dry_rounds = dry_rounds + 1 if not new else 0
    wf.log(f"round {round_index}: {len(new)} new finding(s)")
    return new


wf.loop_until(find_more, lambda _: dry_rounds >= 2, max_iters=6)

report = wf.synthesize(
    sorted(seen),
    prompt="Deduplicate and summarize these findings. Note uncertainty and evidence gaps.",
    model="claude-sonnet-4.5",
    label="report",
)
print(report.content)

File: .copilot/skills/workflow/examples/loop-until-dry.cwf.mjs
// loop-until-dry.cwf.mjs — repeat discovery until two consecutive rounds find nothing new.
export const meta = { name: "loop-until-dry", description: "Iteratively discover findings until the well runs dry, then summarize." };

const question = args || "Find likely issues in this project.";
const seen = new Set();
let dryRounds = 0;

const findMore = async (roundIndex) => {
	const result = await agent(`Find new issues not already seen.\n\nQuestion: ${question}\n\nAlready seen:\n${[...seen].sort()}`, { agentType: "worker", phase: "discover", label: `round-${roundIndex}` });
	const candidates = result.content.split("\n").map((line) => line.trim()).filter(Boolean);
	const fresh = candidates.filter((line) => !seen.has(line));
	fresh.forEach((line) => seen.add(line));
	dryRounds = fresh.length ? 0 : dryRounds + 1;
	log(`round ${roundIndex}: ${fresh.length} new finding(s)`);
	return fresh;
};

await loopUntil(findMore, () => dryRounds >= 2, { maxIters: 6 });

const report = await synthesize([...seen].sort(), { prompt: "Deduplicate and summarize these findings. Note uncertainty and evidence gaps.", label: "report" });
return report.content;
